commit b22f6501bc9838061472128360e0e55d08cb0bd9
Author: Christopher Faulet <cfaulet@haproxy.com>
Date:   Fri Jul 26 14:54:52 2019 +0200

    MINOR: hlua: Don't set request analyzers on response channel for lua actions
    
    Setting some requests analyzers on the response channel was an old trick to be
    sure to re-evaluate the request's analyers after the response's ones have been
    called. It is no more necessary. In fact, this trick was removed in the version
    1.8 and backported up to the version 1.6.
    
    This patch must be backported to all versions since 1.6 to ease the backports of
    fixes on the lua code.
    
    (cherry picked from commit 51fa358432247fe5d7259d9d8a0e08d49d429c73)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/hlua.c b/src/hlua.c
index 21351cd6..36454cdc 100644
--- a/src/hlua.c
+++ b/src/hlua.c
@@ -6873,11 +6873,8 @@ static enum act_return hlua_action(struct act_rule *rule, struct proxy *px,
 		 * is detected on a response channel. This is useful
 		 * only for actions targeted on the requests.
 		 */
-		if (HLUA_IS_WAKERESWR(s->hlua)) {
+		if (HLUA_IS_WAKERESWR(s->hlua))
 			s->res.flags |= CF_WAKE_WRITE;
-			if ((analyzer & (AN_REQ_INSPECT_FE|AN_REQ_HTTP_PROCESS_FE)))
-				s->res.analysers |= analyzer;
-		}
 		if (HLUA_IS_WAKEREQWR(s->hlua))
 			s->req.flags |= CF_WAKE_WRITE;
 		/* We can quit the function without consistency check
